packages/@o3r/testing/schematics/fixture-to-component/index.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  apply,
+  chain,
+  MergeStrategy,
+  mergeWith,
+  move,
+  noop,
+  renameTemplateFiles,
+  Rule,
+  SchematicContext,
+  template,
+  Tree,
+  url
+} from '@angular-devkit/schematics';
+import {
+  applyEsLintFix,
+  getO3rComponentInfo} from '@o3r/schematics';
+import { insertImport } from '@schematics/angular/utility/ast-utils';
+import { applyToUpdateRecorder, Change, InsertChange } from '@schematics/angular/utility/change';
+import { basename, dirname, resolve } from 'node:path';
+import * as ts from 'typescript';
+import type { NgAddFixtureSchematicsSchema } from './schema';
+
+const checkFixture = (componentPath: string, tree: Tree, baseFileName: string) => {
+  const files = [
+    resolve(dirname(componentPath), `./${baseFileName}.fixture.ts`)
+  ];
+  if (files.some((file) => tree.exists(file))) {
+    throw new Error(`Unable to add fixture to this component because it already has at least one of these files: ${files.join(', ')}.`);
+  }
+};
+
+/**
+ * Add fixture to an existing component
+ *
+ * @param options
+ */
+export function ngAddFixture(options: NgAddFixtureSchematicsSchema): Rule {
+  return (tree: Tree, context: SchematicContext) => {
+    const baseFileName = basename(options.path, '.component.ts');
+    const { name } = getO3rComponentInfo(tree, options.path);
+
+    checkFixture(options.path, tree, baseFileName);
+
+    const properties = {
+      ...options,
+      componentFixture: name.concat('Fixture'),
+      name: baseFileName
+    };
+
+    const createFixtureFilesRule: Rule = mergeWith(apply(url('./templates'), [
+      template({ ...properties }),
```

Comment:
do you need to create a copy of `properties` for some reason ?
```suggestion
      template(properties),
```